recipes/deepfplearn/meta.yaml

```diff
@@ -30,6 +34,10 @@ requirements:
     - tensorflow-base
     # - tensorflow-gpu =2
     - wandb =0.12
+    - chemprop
```

Comment:
Ping @soulios

`chemprop` is complicated. `deepfplearn` uses a fork of `chemprop`: https://github.com/yigbt/deepFPlearn/blob/d5a8faa209e8fb8679f3503701f2c716776e8582/setup.py#L34. 

Not sure yet how to handle this. Seems that the `chemprop` maintainers were not keen on integrating the change of the fork.

From a reproducibility point of view, this seems fine. But tricky to maintain. We could just copy the requirements of the chemprop repo to this one, or?

Reply:
for the local env, I changed umap to be installed from pip(because there are a lof of umap packages and that is for sure running).Is there an issue with bioconda to install from pip. My fork of chemprop as well is installed from pip.
The user won't have the latest version of chemprop but they will have sth running.
That is the latest vesion of the environment.yml file.
Is loose versioning an issue?

```
name: dfplenv
channels:
  - conda-forge
  - defaults
dependencies:
  ### dev requirements
  - conda-build>=3.21.8
  - conda>=4.12.0
  - pip>=22.0.4
  - pytest>=7.1.1
  # application requirements
  - jsonpickle>=2.1
  - matplotlib>=3.5.1
  - numpy>=1.19.5
  - pandas>=1.4.2
  - rdkit>=2022.03.1
  - scikit-learn>=1.0.2
  - seaborn>=0.12.2
  - tensorflow>=2.10
  - wandb>=0.12
  - pip:
    - git+https://github.com/soulios/chemprop.git@1d73523e49aa28a90b74edc04aaf45d7e124e338
    - umap>=0.1.1
```

Reply:
> for the local env, I changed umap to be installed from pip(because there are a lof of umap packages and that is for sure running)

I doubt this. You are most likely using 
https://pypi.org/project/umap-learn/ and not 
https://pypi.org/project/umap/.

> Is there an issue with bioconda to install from pip.

Conda python packages are in the end generated by installing the pypi package with pip, but without it's dependencies.

Dependencies are then resolved via conda which does a way better job than pip.

> My fork of chemprop as well is installed from pip.
The user won't have the latest version of chemprop but they will have sth running.

We could do this, I.e the deepfplearn package will contain dfpl itself and the fork of chemprop. But it's kind of a dirty hack. For instance, users could install chemprop itself in the env which would break everything. 

For now there seems to be no way around, but we should talk about a strategy to open PRs for chemprop. 

Thanks for sharing the env, the problem is that in the recipe we can not use pip requirements.